[Clang] Fix name lookup for dependent bases (#114978)
llvm/llvm-project: clang/docs/ReleaseNotes.rst

  by default.
  (`CWG2521: User-defined literals and reserved identifiers <https://cplusplus.github.io/CWG/issues/2521.html>`_).

> - Clang now make correct name lookup when dependent base class is the current instantiation.

Reviewer: ```suggestion
- Fix name lookup for a dependent base class that is the current instantiation.
```

Author: Fixed, thanks

Author: Could we merge this?
